### PR TITLE
Hide `"Show directory"` when `"All combined"` selected

### LINKED ISSUE #7424: Show directory on "All combined" causes `dbg_assert`
image

Happens in both demo browser and editor. Solution would be to not have the "Show directory" button on "All combined".

`src/engine/shared/storage.cpp(764): Type invalid`

### PR DESCRIPTION
**Pull request by dobrykafe**
closes #7424 

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus_demo.cpp [hunk 1]
```diff
@@ -1438,6 +1438,7 @@ void CMenus::RenderDemoBrowserButtons(CUIRect ButtonsView, bool WasListboxItemAc
 	}
 
 	// demos directory button
+	if(m_aCurrentDemoFolder[0] != '\0' || m_DemolistSelectedIndex != 0)
```
Comment (heinrich5991) [inline src/game/client/components/menus_demo.cpp:1441]: This seems unrelated(?)

Comment (dobrykafe) [reply src/game/client/components/menus_demo.cpp:1441]: What does? The issue is present in both the editor and the demo browser.

**Comment by Robyt3** [comment]
Checking the selected demo/map index is brittle and will break if those lists can be sorted. Better check `Item.m_StorageType == IStorage::TYPE_ALL` instead.

**Comment by Robyt3** [comment]
The client also crashes with this assertion when clicking "Show directory" when the `downloadedmaps` or `themes` folder link is selected in the editor.

**Review by Robyt3 — APPROVED** [review]
Thanks.
